libsolidity/codegen/ExpressionCompiler.cpp

```diff
@@ -1035,9 +1035,13 @@ bool ExpressionCompiler::visit(FunctionCall const& _functionCall)
 			break;
 		}
 		case FunctionType::Kind::BlockHash:
+		case FunctionType::Kind::BlobHash:
 		{
 			acceptAndConvert(*arguments[0], *function.parameterTypes()[0], true);
-			m_context << Instruction::BLOCKHASH;
+			if (function.kind() == FunctionType::Kind::BlockHash)
+				m_context << Instruction::BLOCKHASH;
+			else
+				m_context << Instruction::BLOBHASH;
```

Comment:
Not sure I really like this approach - these are separate cases for a reason - why would you combine them and then perform another checker as to what the function kind is? Of course, it's easy to see what's going on once you read the code, but it's confusing at first glance - initially, I thought they may have the same opcode and use the difficulty/prevrandao mechanism based on the fork, which is obviously not the case.

Reply:
Yeah, this one is short enough that it would have been clearer as a separate handler.

Still, when the code is longer this approach makes sense. The handler in `IRGeneratorForStatements.cpp` was actually the opposite case - there merging it with the other cases helps reduce duplication.

Reply:
I initially made them separately but as pointed out here https://github.com/ethereum/solidity/pull/14759#discussion_r1450915633 there are cases that already handle several other built-ins in both `IRGeneratorForStatements` and `ExpressionCompiler`.

Reply:
Which is why I didn't complain about it myself, but @nikola-matic does have a point here :)

Reply:
@r0qs we seem to have wormholed ourselves into a parallel universe where Kamil doesn't complain, and I do. :D